crates/hotshot/src/tasks/mod.rs

```diff
@@ -207,6 +221,42 @@ pub async fn add_consensus_tasks<TYPES: NodeType, I: NodeImplementation<TYPES>>(
     handle.add_task(RewindTaskState::<TYPES>::create_from(&handle).await);
 }
 
+/// Creates a monitor for shutdown events.
+///
+/// # Returns
+/// A `BoxFuture<'static, ()>` that resolves when a `HotShotEvent::Shutdown` is detected.
+///
+/// # Usage
+/// Use in `select!` macros or similar constructs for graceful shutdowns:
+///
+/// ```
+/// let shutdown_signal = handle.create_shutdown_event_monitor();
+/// futures::select! {
+///     _ = shutdown_signal => {
+///         // Perform shutdown logic
+///     },
+///     // ... other branches
+/// }
+/// `
+pub fn create_shutdown_event_monitor<TYPES: NodeType, I: NodeImplementation<TYPES>>(
+    handle: &SystemContextHandle<TYPES, I>,
+) -> BoxFuture<'static, ()> {
+    // Activate the cloned internal event stream
+    let event_stream = handle.internal_event_stream.1.activate_cloned();
+
+    // Create an EventDependency that triggers on HotShotEvent::Shutdown
+    let event_dep = EventDependency::new(
+        event_stream,
+        Box::new(|e| matches!(e.as_ref(), HotShotEvent::Shutdown)),
+    );
+
+    // Create a future that completes when the EventDependency is satisfied
+    async move {
+        let _ = event_dep.completed().await;
+    }
+    .boxed()
```

Comment:
EventDependency is basically just this while loop except it returns the event at the end so the `_` here would alwasy just be `HotShotEvent::Shutdown`   I'd also prefer just doing the loop here since we don't actually need Event returned

Reply:
Thanks @ss-es, @bfish713, Implemented a simple manual loop to check for `HotShotEvent::Shutdown`. No longer relying on `EventDependency`.